[Detections RBAC] Update references to detections RBAC (#3874)
elastic/docs-content: solutions/security/ai/attack-discovery.md

## Role-based access control (RBAC) for Attack Discovery [attack-discovery-rbac]

You need the `Attack Discovery: All` privileges to use Attack Discovery.

> {applies_to}`serverless: ` Your role must also have `Rules: All` privileges to use Attack Discovery.

Reviewer: ```suggestion
{applies_to}`serverless: ` You need the `Rules: All` privileges to use Attack Discovery.
```

> ![attack-discovery-rbac](/solutions/images/security-attck-disc-rbac.png)

Reviewer: ```suggestion
![attack-discovery-rbac](/solutions/images/security-attck-disc-rbac.png "=65%")
```

## Role-based access control (RBAC) for Attack Discovery [attack-discovery-rbac]

You need the `Attack Discovery: All` privileges to use Attack Discovery.

> {applies_to}`serverless: ` You need the `Rules: Read` privilege at a minimum to use Attack Discovery.

Reviewer 2: @yctercero in Serverless, will users need the `Attack Discovery: All` privilege _and_ the `Rules: Read` privilege to use Attack Discover? Or do they only need the `Rules: Read` privilege? The updated screenshot (`security-attck-disc-rbac.png`) makes me think both, but I don't want to assume.

Reviewer 3: Correct, we require both. [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/public/attack_discovery/links.ts#L20) is an example, where we only show Attack Discovery links in the sidebar when you have both `Rules: Read` and `AttackDiscovery`
